staging/src/k8s.io/kubectl/pkg/cmd/apply/prune.go

```diff
@@ -112,10 +117,29 @@ func (p *pruner) prune(namespace string, mapping *meta.RESTMapping) error {
 		if err != nil {
 			return err
 		}
-		annots := metadata.GetAnnotations()
-		if _, ok := annots[corev1.LastAppliedConfigAnnotation]; !ok {
-			// don't prune resources not created with apply
-			continue
+		if p.serverSideApply {
+			managedFields := metadata.GetManagedFields()
+			if managedFields == nil {
+				// don't prune resources not created serversideapply
+				continue
+			}
+			fieldManagerFound := false
+			for _, f := range managedFields {
+				if f.Manager == p.fieldManager {
+					fieldManagerFound = true
+					break
+				}
+			}
+			if !fieldManagerFound {
+				// don't prune resources field manager is different
+				continue
+			}
```

Comment:
[Suggestion] As we are having the same logic here and as well as in `prune.go` under folder `diff`, Can we create a helper function to put the logic at one place? Or is there any specific reason to have same code at two places?

Reply:
Thanks for the suggestion @harry1064